src/Umbraco.Web.UI.Client/src/views/propertyeditors/grid/dialogs/config.controller.js

```diff
@@ -16,7 +16,23 @@ function ConfigController($scope) {
             $scope.model.close();
         }
     }
-    
+
+    $scope.showEmptyState = function() {
+        return $scope.model.config.length === 0
+            && Object.keys($scope.model.config).length === 0
+            && $scope.model.styles.length === 0;
+    }
+
+    $scope.showConfig = function() {
+        return $scope.model.config &&
+            ($scope.model.config.length > 0 || Object.keys($scope.model.config).length > 0);
+    }
+
+    $scope.showStyles = function() {
+        return $scope.model.styles &&
+            ($scope.model.styles.length > 0 || Object.keys($scope.model.styles).length > 0);
+    }
+
```

Comment:
@lars-erik I would probably  use the `vm` syntax here similar to close and submit.
Alternatively it should convert the settings and styles objects to array with a single object.
Not sure if that would cause any issues though.

Reply:
I changed to vm. Also enabled the button.  
I would be super weary of changing the existing code and supported schema beyond this point.  
One broken reference and things may start behaving differently.  
I'd rather see a grid 2.0 instead of meddling too much.

Surely I'm not the only one relying on this schema.